### PR TITLE
MODINV-949 Update Data Import logic to normalize OCLC 035 values

### PR DESCRIPTION
**Pull request by JavokhirAbdullayev**
### Purpose
[MODINV-949](https://folio-org.atlassian.net/browse/MODINV-949)  Update Data Import logic to normalize OCLC 035 values

### Approach
update AdditionalFieldsUtil class

image

### CODE REVIEW

File: src/main/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtil.java [hunk 1]
```diff
@@ -261,6 +272,99 @@ public static void move001To035(Record srcRecord) {
...
+      Set<String> normalized035Subfields = formatOclc(subfields);
+
+      updateOclcSubfield(srsRecord, normalized035Subfields);
+    }
+  }
+
+  public static Set<String> formatOclc(List<Subfield> subFields) {
+    Set<String> processedSet = new LinkedHashSet<>();
+
+    Pattern pattern = Pattern.compile("\\(OCoLC\\)(tfe|ocm|ocn)?0*(\\d+)");
```
Comment (VRohach) [inline src/main/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtil.java:287]: @JavokhirAbdullayev  Let's move it into constant.

File: src/test/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtilTest.java [hunk 2]
```diff
@@ -336,6 +344,69 @@ public void shouldAdd035If001NotEqual003() {
     Assert.assertEquals(expectedParsedContent, parsedRecord.getContent());
   }
 
+  @RunWith(Parameterized.class)
+  public static class OclcFieldNormalizationTest {
```
Comment (VRohach) [inline src/test/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtilTest.java:348]: @JavokhirAbdullayev  Is there any way to add more tests for other corner-cases for these new methods in the AdditionalFieldsUtilTest?

Comment (JavokhirAbdullayev) [reply src/test/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtilTest.java:348]: there should be only one public method, the others should be private, just pushed changes, thanks, nested test class has 4 test cases to check various scenarios

Comment (JavokhirAbdullayev) [reply src/test/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtilTest.java:348]: added additinal tests

**Review by dmytrokrutii — APPROVED** [review]

**Review by VRohach — APPROVED** [review]

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java [hunk 3]
```diff
@@ -111,6 +111,7 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
                 MappingParameters mappingParameters = Json.decodeValue(mappingMetadata.getMappingParams(), MappingParameters.class);
                 AdditionalFieldsUtil.updateLatestTransactionDate(targetRecord, mappingParameters);
                 AdditionalFieldsUtil.move001To035(targetRecord);
+                AdditionalFieldsUtil.normalize035(targetRecord);
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java:114]: this function should also be called when Instance is updated

Comment (JavokhirAbdullayev) [reply src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateInstanceEventHandler.java:114]: good catch, thanks

**Review by KaterynaSenchenko — APPROVED** [review]

File: src/main/java/org/folio/inventory/dataimport/handlers/actions/InstanceUpdateDelegate.java [hunk 4]
```diff
@@ -43,7 +44,7 @@ public Future<Instance> handle(Map<String, String> eventPayload, Record marcReco
     try {
       JsonObject mappingRules = new JsonObject(eventPayload.get(MAPPING_RULES_KEY));
       MappingParameters mappingParameters = new JsonObject(eventPayload.get(MAPPING_PARAMS_KEY)).mapTo(MappingParameters.class);
-
+      AdditionalFieldsUtil.normalize035(marcRecord);
```
Comment (RuslanLavrov) [inline src/main/java/org/folio/inventory/dataimport/handlers/actions/InstanceUpdateDelegate.java:47]: looks like, we need to add this method call to ReplaceInstanceEventHandler for being called when instance is updated during data import
